testsuites/gpu/gpusuite.py

```diff
@@ -0,0 +1,89 @@
+from assertpy import assert_that
+
+from pathlib import Path
+
+from lisa import Environment, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.executable import CustomScript, CustomScriptBuilder
+from lisa.testsuite import simple_requirement
+from lisa.util import commonlib
+from lisa.tools import Lscpu, Lsvmbus
+
+@TestSuiteMetadata(
+    area="gpu",
+    category="functional",
+    description="""
+    This test suite runs the gpu test cases.
+    """,
+    tags=[],
+)
+class gpu(TestSuite):
+
+    unsupported_distro = ["CLEARLINUX", "COREOS"]
+    grid_supported_sku = ["Standard_NV"]
+    cuda_supported_sku = ["Standard_NC", "Standard_ND"]
+        
+    @TestCaseMetadata(
+        description="""
+            This test case verfies kernel support for gpu drivers.
+
+            Steps:
+            1. Install the GPU drivers on the VM.
+            2. Verifies if gpu drivers can be loaded successfully.
+        """,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+        ),
+    )
+    def gpu_validate_gpudriver(self, environment: Environment, node: Node) -> None:
+        # validate if distro is supported for GPU
+        if not validate_gpu_support(self, node):
+            raise SkippedException("distro not supported")
+
+        if any(map((node.AzureNodeSchema.vm_size).__contains__, grid_supported_sku)):
```

Comment:
The `AzureNodeSchema` is runbook configuration, and it may not be specified in requirements. Check it at node context, not schema. It includes runtime information. refer to how `get_node_context` is used in `lisa\sut_orchestrator\azure\features.py`

Reply:
> The `AzureNodeSchema` is runbook configuration, and it may not be specified in requirements. Check it at node context, not schema. It includes runtime information. refer to how `get_node_context` is used in `lisa\sut_orchestrator\azure\features.py`

hmm yeah but node_context seems doesn't have vm_size information
@dataclass
class NodeContext:
    resource_group_name: str = ""
    vm_name: str = ""
    username: str = ""
    password: str = ""
    private_key_file: str = ""

Reply:
> node_runbook = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)

Done